openSUSE build error: undefined reference to `EC_POINT_{set|get}_affine_coordinates_GF2m

Hello,
I get this error by trying to build OpenIKED on openSUSE TW.

```
$ make
[ 22%] Built target compat
[ 23%] Building C object iked/CMakeFiles/iked-shared.dir/ikev2_pld.c.o
[ 25%] Building C object iked/CMakeFiles/iked-shared.dir/imsg_util.c.o
[ 27%] Building C object iked/CMakeFiles/iked-shared.dir/log.c.o
[ 28%] Building C object iked/CMakeFiles/iked-shared.dir/util.c.o
[ 30%] Building C object iked/CMakeFiles/iked-shared.dir/ikev2_map.c.o
[ 32%] Building C object iked/CMakeFiles/iked-shared.dir/eap_map.c.o
[ 35%] Built target iked-shared
[ 37%] Generating parse.c
Scanning dependencies of target iked
[ 38%] Building C object iked/CMakeFiles/iked.dir/ipsec.c.o
[ 40%] Building C object iked/CMakeFiles/iked.dir/pfkey.c.o
[ 42%] Building C object iked/CMakeFiles/iked.dir/ca.c.o
[ 44%] Building C object iked/CMakeFiles/iked.dir/chap_ms.c.o
[ 45%] Building C object iked/CMakeFiles/iked.dir/config.c.o
[ 47%] Building C object iked/CMakeFiles/iked.dir/control.c.o
[ 49%] Building C object iked/CMakeFiles/iked.dir/crypto.c.o
[ 50%] Building C object iked/CMakeFiles/iked.dir/dh.c.o
/home/alex/github/openiked-portable/iked/dh.c: In function ‘ec_point2raw’:
/home/alex/github/openiked-portable/iked/dh.c:618:8: warning: implicit declaration of function ‘EC_POINT_get_affine_coordinates_GF2m’; did you mean ‘EC_POINT_get_affine_coordinates_GFp’? [-Wimplicit-function-declaration]
  618 |   if (!EC_POINT_get_affine_coordinates_GF2m(ecgroup,
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |        EC_POINT_get_affine_coordinates_GFp
/home/alex/github/openiked-portable/iked/dh.c: In function ‘ec_raw2point’:
/home/alex/github/openiked-portable/iked/dh.c:686:8: warning: implicit declaration of function ‘EC_POINT_set_affine_coordinates_GF2m’; did you mean ‘EC_POINT_set_affine_coordinates_GFp’? [-Wimplicit-function-declaration]
  686 |   if (!EC_POINT_set_affine_coordinates_GF2m(ecgroup,
      |        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |        EC_POINT_set_affine_coordinates_GFp
[ 52%] Building C object iked/CMakeFiles/iked.dir/eap.c.o
[ 54%] Building C object iked/CMakeFiles/iked.dir/iked.c.o
[ 55%] Building C object iked/CMakeFiles/iked.dir/ikev2.c.o
[ 57%] Building C object iked/CMakeFiles/iked.dir/ikev2_msg.c.o
[ 59%] Building C object iked/CMakeFiles/iked.dir/ocsp.c.o
[ 61%] Building C object iked/CMakeFiles/iked.dir/policy.c.o
[ 62%] Building C object iked/CMakeFiles/iked.dir/proc.c.o
[ 64%] Building C object iked/CMakeFiles/iked.dir/smult_curve25519_ref.c.o
[ 66%] Building C object iked/CMakeFiles/iked.dir/timer.c.o
[ 67%] Building C object iked/CMakeFiles/iked.dir/parse.c.o
[ 69%] Linking C executable iked
/usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/iked.dir/dh.c.o: in function `ec_point2raw':
dh.c:(.text+0x7cb): undefined reference to `EC_POINT_get_affine_coordinates_GF2m'
/usr/lib64/gcc/x86_64-suse-linux/10/../../../../x86_64-suse-linux/bin/ld: CMakeFiles/iked.dir/dh.c.o: in function `ec_raw2point':
dh.c:(.text+0xa7e): undefined reference to `EC_POINT_set_affine_coordinates_GF2m'
collect2: error: ld returned 1 exit status
make[2]: *** [iked/CMakeFiles/iked.dir/build.make:400: iked/iked] Error 1
make[1]: *** [CMakeFiles/Makefile2:202: iked/CMakeFiles/iked.dir/all] Error 2
make: *** [Makefile:149: all] Error 2

```

changing these to *_GFp fixed the build problem. The question is: do we need this patch for openSUSE? Does it works on other GNU systems like Debian?

--- a/iked/dh.c
+++ b/iked/dh.c
@@ -609,16 +609,9 @@ ec_point2raw(struct dh_group *group, const EC_POINT *point,
 	if ((ecgroup = EC_KEY_get0_group(group->ec)) == NULL)
 		goto done;
 
-	if (EC_METHOD_get_field_type(EC_GROUP_method_of(ecgroup)) ==
-	    NID_X9_62_prime_field) {
-		if (!EC_POINT_get_affine_coordinates_GFp(ecgroup,
-		    point, x, y, bnctx))
-			goto done;
-	} else {
-		if (!EC_POINT_get_affine_coordinates_GF2m(ecgroup,
-		    point, x, y, bnctx))
-			goto done;
-	}
+	if (!EC_POINT_get_affine_coordinates_GFp(ecgroup,
+	    point, x, y, bnctx))
+		goto done;
 
 	xoff = xlen - BN_num_bytes(x);
 	bzero(buf, xoff);
@@ -677,16 +670,9 @@ ec_raw2point(struct dh_group *group, uint8_t *buf, size_t len)
 	if ((point = EC_POINT_new(ecgroup)) == NULL)
 		goto done;
 
-	if (EC_METHOD_get_field_type(EC_GROUP_method_of(ecgroup)) ==
-	    NID_X9_62_prime_field) {
-		if (!EC_POINT_set_affine_coordinates_GFp(ecgroup,
-		    point, x, y, bnctx))
-			goto done;
-	} else {
-		if (!EC_POINT_set_affine_coordinates_GF2m(ecgroup,
-		    point, x, y, bnctx))
-			goto done;
-	}
+	if (!EC_POINT_set_affine_coordinates_GFp(ecgroup,
+	    point, x, y, bnctx))
+		goto done;
 
 	ret = 0;
  done:

